src/include/msvc.h

```diff
@@ -77,6 +77,21 @@ __wt_atomic_cas_ptr(void *vp, void *old_val, void *new_val)
               (volatile __int64 *)vp, (int64_t)new_val, (int64_t)old_val) == ((int64_t)old_val));
 }
 
+/*
+ * __wt_atomic_load_acquire64 --
+ *     Read a memory value with atomic acquire synchronization.
+ */
+static inline void
+__wt_atomic_load_acquire64(void *ret, void *ptr)
+{
+    /*
+     * Since we only support x86 on windows and x86 has Total Storage Ordering, no actual read
+     * barrier is needed here. We only need to ensure the compiler is not reordering the read by
+     * casting the pointer to a volatile pointer.
+     */
+    *(volatile void **)ret = *(void **)ptr;
```

Comment:
Suppose the intention is only to stop the compiler from reordering. In that case, we should issue a ```compiler only  barrier```, which is abstracted out from platform implementations as ```WT_BARRIER```. With MSVC it translates into ```_ReadWriteBarrier```:
https://learn.microsoft.com/en-us/cpp/intrinsics/readwritebarrier?view=msvc-170

The ```WT_BARRIER``` implementation is just below this new function.

Reply:
Thanks